flutter_local_notifications_windows/lib/src/plugin/base.dart

```diff
@@ -0,0 +1,67 @@
+import "package:flutter_local_notifications_platform_interface/flutter_local_notifications_platform_interface.dart";
+import "package:timezone/timezone.dart";
+
+import "../details.dart";
+
+export "package:flutter_local_notifications_platform_interface/flutter_local_notifications_platform_interface.dart";
+export "package:timezone/timezone.dart";
+
+/// The Windows implementation of `package:flutter_local_notifications`.
+abstract class WindowsNotificationsBase extends FlutterLocalNotificationsPlatform {
+  /// Initializes the plugin. No other method should be called before this.
+  Future<bool> initialize(
+    WindowsInitializationSettings settings, {
+    DidReceiveNotificationResponseCallback? onNotificationReceived,
+  });
+
+  /// Releases any resources used by this plugin.
+  void dispose();
+
+  /// The raw XML passed to the Windows API.
+  ///
+  /// See https://learn.microsoft.com/en-us/uwp/schemas/tiles/toastschema/schema-root.
+  /// For validation, see [the Windows Notifications Visualizer](https://learn.microsoft.com/en-us/windows/apps/design/shell/tiles-and-notifications/notifications-visualizer).
+  Future<void> showRawXml({
+    required int id,
+    required String xml,
+    Map<String, String> bindings = const {},
+  });
+
+  @override
+  Future<void> show(
+    int id,
+    String? title,
+    String? body, {
+    String? payload,
+    WindowsNotificationDetails? details,
+  });
+
+  /// Schedules a notification to appear at the given date and time.
+  Future<void> zonedSchedule(
+    int id,
+    String? title,
+    String? body,
+    TZDateTime scheduledDate,
+    WindowsNotificationDetails? details, {
+    String? payload,
+  });
+
+  /// Updates the progress bar in the notification with the given ID.
+  ///
+  /// Note that in order to update [WindowsProgressBar.label], it must
+  /// not have been set to null when [show] was called.
```

Comment:
Presumably the a notification with a progress bar can be scheduled to be shown e.g. using `zonedSchedule()` and that it doesn't require the notification to be shown immediately via `show()`

Reply:
I thought about that, but I figured it wouldn't make much sense to update a progress bar that was scheduled to be shown, because then you can't actually be sure you're updating anything at all. Besides, a progress bar is meant to represent real work done in real-time, not predicted to occur later. So it seemed to me that the only reasonable time to update a progress bar would be right after showing it,

Reply:
Changed the comment